extras/devices/LVFS/tests/test_LVFS.py

```diff
@@ -0,0 +1,66 @@
+import unittest
+import json
+from devices import *
+from unittest.mock import patch
+from devices.LVFS.tests import fwupd_data
+
+device_results = json.loads(fwupd_data.get_results)
+
+
+class TestLVFSDevice(unittest.TestCase):
+    def mock_run_cmd(*args, **kwargs):
+        if "sudo fwupdmgr get-results" in args[1]:
+            return 0, json.dumps(device_results), ""
+        return 0, "", ""
+
+    # current System Firmware version is the previous version
+    def test_upgradable(self):
+        with patch("devices.LVFS.LVFS.LVFSDevice.run_cmd") as mock_path:
+            mock_path.side_effect = self.mock_run_cmd
+            device = LVFSDevice("", "", "")
+            print(type(fwupd_data.get_device))
+            device._parse_fwupd_raw(fwupd_data.get_device)
+            self.assertTrue(device.upgrade())
+            self.assertFalse(device.downgrade())
+
+    # current System Firmware version is the latest version
+    def test_downgradable(self):
+        with patch("devices.LVFS.LVFS.LVFSDevice.run_cmd") as mock_path:
+            mock_path.side_effect = self.mock_run_cmd
+            device = LVFSDevice("", "", "")
+
+            device_info = json.loads(fwupd_data.get_device)
+            device_info["Devices"][5]["Version"] = "2.91"
+            device_info["Devices"][5]["Releases"][0]["Flags"] = []
+            device_info["Devices"][5]["Releases"][1]["Flags"] = [
+                "is-downgrade"
+            ]
+            device._parse_fwupd_raw(json.dumps(device_info))
+            self.assertFalse(device.upgrade())
+            self.assertTrue(device.downgrade())
+
+    def test_results(self):
+        global device_results
```

Comment:
IIRC, we use pytest in this project (unittest classes are supported but the tests are executed through pytest)
If so, it's a good practice to leave a doc-string for each testcase and describe what you test and expected results. This is because when you get an error during running tests, pytest output the description for each failed test (using these doc-strings)

Reply:
Done. Doc-strings added.